src/App.js

```diff
@@ -20,7 +20,7 @@ function App() {
     filterHandler();
     saveLocalTodos();
     saveLocalDetails();
-  }, [todos, status]);
+  }, [todos, status, details]);
```

Comment:
this hook was not executing due to absent dependency, always check whether all dependencies are added whose changes shall execute the effect functions. @raiab